awscrt/s3.py

```diff
@@ -454,6 +454,34 @@ def cancel(self):
         _awscrt.s3_meta_request_cancel(self)
 
 
+class S3ResponseError(awscrt.exceptions.AwsCrtError):
+    '''
+    An error response from S3.
+
+    Subclasses :class:`awscrt.exceptions.AwsCrtError`.
+
+    Attributes:
+        status_code (int): HTTP response status code.
+        headers (list[tuple[str, str]]): Headers from HTTP response.
+        body (Optional[bytes]): Body of HTTP response (if any).
```

Comment:
trivial: We can add more details indicating it is in XML, and that it can be empty/null for cases like HeadRequest where S3 doesn't give us anything.